llvm/lib/SYCLLowerIR/LocalAccessorToSharedMemory.cpp

```diff
@@ -242,89 +210,13 @@ class LocalAccessorToSharedMemory : public ModulePass {
     return NF;
   }
 
-  void populateCudaKernels(Module &M, SmallVector<KernelPayload> &Kernels) {
-    // Access `nvvm.annotations` to determine which functions are kernel entry
-    // points.
-    auto *NvvmMetadata = M.getNamedMetadata("nvvm.annotations");
-    if (!NvvmMetadata)
-      return;
-
-    // It is possible that the annotations node contains multiple pointers to
-    // the same metadata, recognise visited ones.
-    SmallSet<MDNode *, 4> Visited;
-    for (auto *MetadataNode : NvvmMetadata->operands()) {
-      if (Visited.contains(MetadataNode) || MetadataNode->getNumOperands() != 3)
-        continue;
-
-      Visited.insert(MetadataNode);
-
-      // NVPTX identifies kernel entry points using metadata nodes of the form:
-      //   !X = !{<function>, !"kernel", i32 1}
-      const MDOperand &TypeOperand = MetadataNode->getOperand(1);
-      auto *Type = dyn_cast<MDString>(TypeOperand);
-      if (!Type)
-        continue;
-      // Only process kernel entry points.
-      if (Type->getString() != "kernel")
-        continue;
-
-      // Get a pointer to the entry point function from the metadata.
-      const MDOperand &FuncOperand = MetadataNode->getOperand(0);
-      if (!FuncOperand)
-        continue;
-      auto *FuncConstant = dyn_cast<ConstantAsMetadata>(FuncOperand);
-      if (!FuncConstant)
-        continue;
-      auto *Func = dyn_cast<Function>(FuncConstant->getValue());
-      if (!Func)
-        continue;
-
-      Kernels.push_back(KernelPayload(Func, MetadataNode));
-    }
-  }
-
-  void populateAMDKernels(Module &M, SmallVector<KernelPayload> &Kernels) {
-    for (auto &F : M) {
-      if (F.getCallingConv() == CallingConv::AMDGPU_KERNEL)
-        Kernels.push_back(KernelPayload(&F));
-    }
-  }
-
-  void populateKernels(Module &M, SmallVector<KernelPayload> &Kernels,
-                       ArchType AT) {
-    switch (AT) {
-    case ArchType::Cuda:
-      return populateCudaKernels(M, Kernels);
-    case ArchType::AMDHSA:
-      return populateAMDKernels(M, Kernels);
-    default:
-      llvm_unreachable("Unsupported arch type.");
-    }
-  }
-
-  void postProcessCudaKernels(
+  void postProcessKernels(
       SmallVector<std::pair<Function *, KernelPayload>> &NewToOldKernels) {
```

Comment:
```suggestion
      SmallVectorImpl<std::pair<Function *, KernelPayload>> &NewToOldKernels) {
```

Reply:
And the corresponding change in `LocalAccessorToSharedMemory.h` can be required.